### ISSUE TITLE
failed to parse empty duration when creating a single experiment

### ISSUE THREAD

**Issue by cwen0** [chaos-mesh/chaos-mesh#2452]
## Bug Report

**What version of Kubernetes are you using?**
 

```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.7", GitCommit:"132a687512d7fb058d0f5890f07d4121b3f0a2e2", GitTreeState:"clean", BuildDate:"2021-05-12T12:40:09Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.7", GitCommit:"132a687512d7fb058d0f5890f07d4121b3f0a2e2", GitTreeState:"clean", BuildDate:"2021-05-12T12:32:49Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"linux/arm64"}
```

**What version of Chaos Mesh are you using?**

latest 
```
Controller manager Version: version.Info{GitVersion:"master-gd0739a11c8e184", GitCommit:"d0739a11c8e184bdbf4fd0bd35bf5e213bb66866", BuildDate:"2021-11-02T10:48:14Z", GoVersion:"go1.16.7", Compiler:"gc", Platform:"linux/arm64"}
```

**What did you do?**
 

Create a Networchaos experiment without duration on Chaos Dashboard 

![image](https://user-images.githubusercontent.com/22956341/139835440-a4020f62-fae4-4895-93be-0e48d7f99dfb.png)

```yaml
kind: NetworkChaos
apiVersion: chaos-mesh.org/v1alpha1
metadata:
  namespace: default
  name: del-xx
  annotations:
    experiment.chaos-mesh.org/pause: 'true'
spec:
  selector:
    namespaces:
      - default
    labelSelectors:
      app: web-show
  mode: one
  action: delay
  duration: ''
  delay:
    latency: 10ms
    correlation: '0'
    jitter: 0ms
  direction: to
```

**What did you expect to see?** 

All is OK!

**What did you see instead?** 

Controller manager return an error: 

```
2021-11-02T10:57:06.036Z        INFO    condition       updating conditions     {"conditions": [{"type":"Paused","status":"False","reason":""},{"type":"Selected","status":"True","reason":""},{"type":"AllInjected","status":"False","reason":""},{"type":"AllRecovered","status":"False
","reason":""}]}
2021-11-02T10:57:06.036Z        INFO    records iterating record        {"record": {"id":"default/web-show-6fb4798b4c-6vvh5","selectorKey":".","phase":"Not Injected/Wait"}, "desiredPhase": "Run"}
2021-11-02T10:57:06.036Z        INFO    records apply chaos     {"id": "default/web-show-6fb4798b4c-6vvh5"}
2021-11-02T10:57:06.036Z        INFO    trafficcontrol  traffic control Apply   {"namespace": "default", "name": "del-xx"}
2021-11-02T10:57:06.036Z        INFO    records iterating record        {"record": {"id":"default/web-show-6fb4798b4c-6vvh5","selectorKey":".","phase":"Not Injected/Wait"}, "desiredPhase": "Run"}
2021-11-02T10:57:06.043Z        INFO    records apply chaos     {"id": "default/web-show-6fb4798b4c-6vvh5"}
2021-11-02T10:57:06.043Z        INFO    trafficcontrol  traffic control Apply   {"namespace": "default", "name": "del-xx"}
2021-11-02T10:57:06.036Z        ERROR   desiredphase    failed to parse duration        {"error": "time: invalid duration \"\""}
github.com/chaos-mesh/chaos-mesh/controllers/desiredphase.(*reconcileContext).Reconcile
        /mnt/controllers/desiredphase/controller.go:128
github.com/chaos-mesh/chaos-mesh/controllers/desiredphase.(*Reconciler).Reconcile
        /mnt/controllers/desiredphase/controller.go:64
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:298
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:253
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:214
2021-11-02T10:57:06.043Z        INFO    desiredphase    modify desiredPhase     {"desiredPhase": "Run"}
2021-11-02T10:57:06.051Z        INFO    records iterating record        {"record": {"id":"default/web-show-6fb4798b4c-6vvh5","selectorKey":".","phase":"Not Injected/Wait"}, "desiredPhase": "Run"}
2021-11-02T10:57:06.051Z        INFO    records apply chaos     {"id": "default/web-show-6fb4798b4c-6vvh5"}
2021-11-02T10:57:06.051Z        INFO    trafficcontrol  traffic control Apply   {"namespace": "default", "name": "del-xx"}
2021-11-02T10:57:06.051Z        ERROR   desiredphase    failed to parse duration        {"error": "time: invalid duration \"\""}
github.com/chaos-mesh/chaos-mesh/controllers/desiredphase.(*reconcileContext).Reconcile
        /mnt/controllers/desiredphase/controller.go:128
github.com/chaos-mesh/chaos-mesh/controllers/desiredphase.(*Reconciler).Reconcile
        /mnt/controllers/desiredphase/controller.go:64
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:298
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:253
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:214
2021-11-02T10:57:06.051Z        INFO    desiredphase    modify desiredPhase     {"desiredPhase": "Run"}
```

**Output of chaosctl**

**Comment by STRRL** [comment]
I can not reproduce it from neither `d0739a11c8e184` nor current master `10addb4e3bd7`, with neither `kubectl -f <chaos>.yaml` and using chaos dashboard. 🥲🥲🥲

More information is required.

**Comment by cwen0** [comment]
@STRRL Thanks for your reply! I found this issue has been fixed in the latest version. Just a suggestion, If we don't set `duration`, we can delete this field on the YAML file directly instead of setting it empty.  
![image](https://user-images.githubusercontent.com/22956341/140451781-5dfb22c1-4122-4729-9530-1c4a8839b1bb.png)

@g1eny0ung PTAL

**Comment by g1eny0ung** [comment]
This was fixed in #2511. All empty fields are now removed before committing.
